src/StreamJsonRpc/JsonMessageFormatter.cs

```diff
@@ -583,17 +583,55 @@ internal JsonRpcRequest(JsonMessageFormatter formatter)
 
             public override ArgumentMatchResult TryGetTypedArguments(ReadOnlySpan<ParameterInfo> parameters, Span<object> typedArguments)
             {
-                // Special support for accepting a single JToken instead of all parameters individually.
-                if (parameters.Length == 1 && parameters[0].ParameterType == typeof(JToken) && this.NamedArguments != null)
+                if (parameters.Length == 1 && this.NamedArguments != null)
                 {
-                    var obj = new JObject();
-                    foreach (KeyValuePair<string, object> property in this.NamedArguments)
+                    // Special support for accepting a single JToken instead of all parameters individually.
+                    if (parameters[0].ParameterType == typeof(JToken))
                     {
-                        obj.Add(new JProperty(property.Key, property.Value));
+                        var obj = new JObject();
+                        foreach (KeyValuePair<string, object> property in this.NamedArguments)
+                        {
+                            obj.Add(new JProperty(property.Key, property.Value));
+                        }
+
+                        typedArguments[0] = obj;
+                        return ArgumentMatchResult.Success;
                     }
 
-                    typedArguments[0] = obj;
-                    return ArgumentMatchResult.Success;
+                    // Check if it is constructable with default constructor
+                    if (parameters[0].ParameterType.GetConstructor(Type.EmptyTypes) != null && !parameters[0].ParameterType.IsAbstract)
```

Comment:
I think this is a breaking change, since any method with just one parameter whose type as a default constructor would not be interpreted differently yet it was a valid scenario before this change. It also deviates from the JSON-RPC spec. Given those, I feel like this should be an explicit opt-in. Our `JsonRpcMethodAttribute` might be the best way to do this. Suppose we add a property to that attribute to indicate that we want the method's one parameter to receive the deserialized result of the entire parameter object? 
The server might also return an error to the client if it doesn't use parameter object syntax in this case, since an array of values can't be reliably matched to named members on the CLR object.

Reply:
I added a property on `JsonMethodAttribute` to indicate whether the method supports this or not.